### DISCUSSION TITLE
Unauthorized: Access to Request Resource is Denied. While Hitting the Finance API

### DISCUSSION THREAD

**Discussion by Priyankk18k** [saleweaver/python-amazon-sp-api#137] [Q&A] [answered]
Hi,
I'm able to get the Orders from the Orders API but I'm Not able to get the Finances from the API. 
The Code I'm Using is:- 

finance = Finances(credentials=client_config, marketplace=Marketplaces.ES)
finance.get_financial_events_for_order(order_id='111-111111-1111111',MarketplaceId=marketplaces)

I'm Getting this as a Response 
SellingApiForbiddenException: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]
This Same Error Was coming when I was using the Get Orders because I was not sending the Marketplaces. Now the Same Error Comes When I try to fetch The finances. can anyone help, please?

**Comment by Priyankk18k** [comment, marked as answer]
I solved this Error. 
If you are facing the same issue go to developer Profile and See If The API you are hitting has Access to view the Information or not. In my case, It was not. Now check all the boxes you need and then register(save and exit). After this got to edit the app and select the Boxes you checked in the developer form. Now generate new Refresh Token for the app. after this Paste this in your code and it will work. Check The marketplace. If you are passing the Wrong marketplace it will throw the Same access denied message.
Thanks.

**Reply by mk-hasan** [reply]
Hi can you please tell me how did you get the order data? I meant which credentials i need to provide? 
I am confused.

Thanks

**Comment by Priyankk18k** [comment]
You need to provide
Acces key and secret key for your IAM ACCOUNT. And rest will be created
whne you make an app.

**Reply by mk-hasan** [reply]
I was suing MWS API and could get the report data using Seller ID, Access Key and Secret Key. 
What is the difference with SP API, can you please tell me?

**Comment by Priyankk18k** [comment]
I don't know about the actual work of MWS because I worked directly with
SP-API. So In SP-API They have changed the format of the output and
provided more functionalities. In SP-API You have to first create the App
that the app will be responsible to restrict a developer for certain Data.
You need to fill the form of the Developer in which you will define for
which articular thing you need access. Then After this you need to go
inside the app and select those things(boxes) which you selected while
filling the form. So after this, you can access the Data. Every app has its
creds that will be passed in the API call
